tests/safety/test_gm.py

```diff
@@ -90,6 +94,18 @@ def _torque_cmd_msg(self, torque, steer_req=1):
     values = {"LKASteeringCmd": torque}
     return self.packer.make_can_msg_panda("ASCMLKASteeringCmd", 0, values)
 
+  def test_prev_brake(self, test="brake"):
+    super().test_prev_brake(test="brake")
+    super().test_prev_brake(test="regen")
```

Comment:
these tests should be common

Reply:
What do you want `_user_regen_msg` to return for the other cars?

Reply:
Maybe that should be split up so the common test tests behavior using the state of the variable, and then there's a car-specific test for GM that tests that the CAN message affects the state of the variable.

This sounds like a good clean up for all of the tests

Reply:
This now tests all previous brake tests for all safety models, and only the regen tests for safety models that implement `_user_regen_msg` (it's impossible to test anything without it)

Reply:
Just made it detect the definition of the regen message, as there's nothing to test if the safety model test does not use it